text/0000-default-field-values.md

```diff
@@ -0,0 +1,1658 @@
+enum FileFavor {
+    #[default]
+    Normal,
+    Ours,
+    Theirs,
+    Union,
+}
+```
+
+## Clearer documentation and more local reasoning
+
+Providing good defaults when such exist is part of any good design that makes
+a physical tool, UI design, or even data-type more ergonomic and easily usable.
+However, that does not mean that the defaults provided can just be ignored
+and that they need not be understood. This is especially the case when you
+are moving away from said defaults and need to understand what they were.
+Furthermore, it is not too uncommon to see authors writing in the documentation
+of a data-type that a certain value is the default.
+
+All in all, the defaults of a data-type are therefore important properties.
+By encoding the defaults right where the data-type is defined gains can be made
+in terms of readability particularly wrt. the *skimmability* of code.
+In particular, it is easier to see what the defaults are if you can directly
+look at the `rustdoc` page and read:
+
+```rust
+#[derive(Default)]
+enum Foo {
+    #[default]
+    Bar {
+        alpha: u8 = 42,
+    },
+    Baz {
+        beta: u16 = 24,
+        gamma: bool,
+    }
+}
+```
+
+This way, you do not need to open up the code of the `Default` implementation
+to see what the values encoded are.
+
+## Usage by other `#[derive(..)]` macros
+
+[`serde`]: https://serde.rs/attributes.html
+
+Custom derive macros exist that have a notion of or use default values.
+
+### `serde`
+
+For example, the [`serde`] crate provides a `#[serde(default)]` attribute that
+can be used on `struct`s, and fields. This will use the field's or type's
+`Default` implementations. This works well with field defaults; `serde` can
+either continue to rely on `Default` implementations in which case this RFC
+facilitates specification of field defaults; or it can directly use the default
+values provided in the type definition.
+
+### `structopt`
+
+Another example is the `structopt` crate with which you can write:
+
+```rust
+#[derive(Debug, StructOpt)]
+#[structopt(name = "example", about = "An example of StructOpt usage.")]
+struct Opt {
+    /// Set speed
+    #[structopt(short = "s", long = "speed", default_value = "42")]
+    speed: f64,
+    ...
+}
+```
+
+By having default field values in the language, `structopt` could let you write:
+
+```rust
+#[derive(Debug, StructOpt)]
+#[structopt(name = "example", about = "An example of StructOpt usage.")]
+struct Opt {
+    /// Set speed
+    #[structopt(short = "s", long = "speed")]
+    speed: f64 = 42,
+    ...
+}
+```
+
+### `derive_builder`
+
+[`derive_builder`]: https://docs.rs/derive_builder/0.7.0/derive_builder/#default-values
+
+A third example comes from the crate [`derive_builder`]. As the name implies,
+you can use it to `#[derive(Builder)]`s for your types. An example is:
+
+```rust
+#[derive(Builder, Debug, PartialEq)]
+struct Lorem {
+    #[builder(default = "42")]
+    pub ipsum: u32,
+}
+```
+
+### `proptest_derive`
```

Comment:
hm... this example seems more questionable to me, and I'm not sure it adds anything the others didn't cover. I'd personally choose to remove it, but up to you

Reply:
I'd tend to agree with Taylor here.